demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,552 @@
+                              void * pBuffer,
+                              size_t bytesToRecv );
+
+/**
+ * @brief Send an HTTP request based on a specified method and path.
+ *
+ * @param[in] pTransportInterface The transport interface for making network calls.
+ * @param[in] pHost The host name of the server.
+ * @param[in] pMethod The HTTP request method.
+ * @param[in] pPath The Request-URI to the objects of interest.
+ *
+ * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
+ */
+static int sendHttpRequest( HTTPTransportInterface_t * pTransportInterface,
+                            const char * pHost,
+                            const char * pMethod,
+                            const char * pPath );
+
+/*-----------------------------------------------------------*/
+
+static int connectToServer( const char * pServer,
+                            uint16_t port,
+                            int * pTcpSocket )
+{
+    int returnStatus = EXIT_SUCCESS;
+    struct addrinfo hints, * pIndex, * pListHead = NULL;
+    struct sockaddr * pServerInfo;
+    uint16_t netPort = htons( port );
+    socklen_t serverInfoLength;
+    struct timeval transportTimeout;
+
+    /* Add hints to retrieve only TCP sockets in getaddrinfo. */
+    ( void ) memset( &hints, 0, sizeof( hints ) );
+    /* Address family of either IPv4 or IPv6. */
+    hints.ai_family = AF_UNSPEC;
+    /* TCP Socket. */
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_protocol = IPPROTO_TCP;
+
+    /* Perform a DNS lookup on the given host name. */
+    returnStatus = getaddrinfo( pServer, NULL, &hints, &pListHead );
+
+    if( returnStatus != -1 )
+    {
+        LogInfo( ( "Performing DNS lookup on %s.",
+                   SERVER_HOST ) );
+
+        /* Attempt to connect to one of the retrieved DNS records. */
+        for( pIndex = pListHead; pIndex != NULL; pIndex = pIndex->ai_next )
+        {
+            *pTcpSocket = socket( pIndex->ai_family, pIndex->ai_socktype, pIndex->ai_protocol );
+
+            if( *pTcpSocket == -1 )
+            {
+                continue;
+            }
+
+            pServerInfo = pIndex->ai_addr;
+
+            if( pServerInfo->sa_family == AF_INET )
+            {
+                /* IPv4 */
+                ( ( struct sockaddr_in * ) pServerInfo )->sin_port = netPort;
+                serverInfoLength = sizeof( struct sockaddr_in );
+                inet_ntop( pServerInfo->sa_family,
+                           &( ( struct sockaddr_in * ) pServerInfo )->sin_addr,
+                           resolvedIpAddr,
+                           sizeof( resolvedIpAddr ) );
+            }
+            else
+            {
+                /* IPv6 */
+                ( ( struct sockaddr_in6 * ) pServerInfo )->sin6_port = netPort;
+                serverInfoLength = sizeof( struct sockaddr_in6 );
+                inet_ntop( pServerInfo->sa_family,
+                           &( ( struct sockaddr_in * ) pServerInfo )->sin_addr,
```

Comment:
Should this be cast to `struct sockaddr_in6`?
Has this been tested against an IPV6 address?

Reply:
Good catch